src/providers/on-chain-quote-provider.ts

```diff
@@ -270,15 +287,19 @@ export class V3QuoteProvider implements IV3QuoteProvider {
       baseBlockOffset: 0,
       rollback: { enabled: false },
     },
+    protected isMixedRouteQuoteProvider: boolean = false,
```

Comment:
Chose to add this extra parameter here to allow us to toggle between which lookup table we use for the Quoter contract addresses, since we can't use the generic type `TRoute` as it's only available at runtime.